src/figuresClasses.ts

```diff
@@ -1,11 +1,99 @@
-export interface Figure {}
+export interface Figure {
+  shape: 'triangle' | 'circle' | 'rectangle';
+  color: 'red' | 'green' | 'blue';
+  getArea: () => number;
+}
+
+export class Triangle implements Figure {
+  shape: 'triangle';
+
+  color: 'red' | 'green' | 'blue';
+
+  a: number;
+
+  b: number;
+
+  c: number;
+
+  constructor(
+    color: 'red' | 'green' | 'blue',
+    a: number,
+    b: number,
+    c: number,
+  ) {
+    if (a <= 0 || b <= 0 || c <= 0) {
+      throw new Error('Triangle side lengths must be positive');
+    }
+
+    if (a >= b + c || b >= a + c || c >= a + b) {
+      throw new Error('Triangle inequality violated sum must be greater sum');
+    }
+
+    this.shape = 'triangle';
+    this.color = color;
+    this.a = a;
+    this.b = b;
+    this.c = c;
+  }
+
+  getArea(): number {
+    const s = (this.a + this.b + this.c) / 2;
```

Comment:
Always try give clarify name for variables.
```suggestion
    const semiPerimeter = (this.a + this.b + this.c) / 2;
```